### PR TITLE
#19 - Added Versions.all() method

### PR DESCRIPTION
**Pull request by olegmoz**
Part of issue #19 
Listing of versions is required to generate package metadata. All package versions are listed ordered

### CODE REVIEW

**Comment by paulodamaso** [comment]
@g4s8 Please assign me

**Comment by 0crat** [comment]
This pull request [#59](https://github.com/artipie/nuget-adapter/issues/59) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/db4aa02d-8a67-4d27-83ce-21662893132e); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artpie/nuget/Versions.java [hunk 1]
```diff
@@ -101,6 +104,23 @@ public Versions add(final Version version) throws IOException {
         );
     }
 
+    /**
+     * Read all package versions.
+     *
+     * @return All versions sorted by natural order.
+     * @throws IOException In case of any I/O problems.
+     */
+    public List<Version> all() throws IOException {
```
Comment (Vatavuk) [inline src/main/java/com/artpie/nuget/Versions.java]: @olegmoz We are returning a mutable list. Either change method signature to return an `Iterable` or return an immutable list.

Comment (olegmoz) [reply src/main/java/com/artpie/nuget/Versions.java]: @Vatavuk I'd prefer to return `List` here, to emphasize that it is ordered and finite. And do not see any problem with returning a mutable instance: it is a copy for every method call and changes to one list won't affect others.

Comment (Vatavuk) [reply src/main/java/com/artpie/nuget/Versions.java]: @olegmoz but we can mutate the returned `List` itself on the higher levels. I would suggest then that we use `collect(ImmutableList.toImmutableList())`

Comment (olegmoz) [reply src/main/java/com/artpie/nuget/Versions.java]: @Vatavuk done

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @vatavuk[/z](https://www.0crat.com/u/vatavuk)[/z](https://www.0crat.com/u/vatavuk) didn't make CR comments

**Comment by 0crat** [comment]
The project doesn't have enough funds, can't make a payment

**Pull request merged**